python/console/console.py

```diff
@@ -37,22 +37,64 @@
 
 import sys
 
-_console = None
+_console = None         # current console widget container
+_consoleEditor = None   # console widget
+_consoleAsDock = None   # status: True=dockwidget, False=top-level window
 
 
 def show_console():
     """ called from QGIS to open the console """
     global _console
-    if _console is None:
-        parent = iface.mainWindow() if iface else None
-        _console = PythonConsole(parent)
+    global _consoleEditor
+    global _consoleAsDock
+
+    s = QgsSettings()
+
+    # get the setting as a bool value (for "true"/"false" values QgsSettings returns a string to Python)
+    dockValue = QVariant(s.value("qgis/dockPythonConsole","true"))
```

Comment:
You didn't run scripts/prepare-commit.sh ;) There is a missing space after the comma - although I don't think the failing test is related to your commit.